repo/content/content_manager_indexes.go

```diff
@@ -87,48 +102,52 @@ func (bm *Manager) compactAndDeleteIndexBlobs(ctx context.Context, indexBlobs []
 		return nil
 	}
 
-	formatLog(ctx).Debugf("compacting %v contents", len(indexBlobs))
+	formatLog(ctx).Debugf("compacting %v index blobs", len(indexBlobs))
 
-	t0 := time.Now() // allow:no-inject-time
 	bld := make(packIndexBuilder)
 
+	var inputs, outputs []blob.Metadata
+
 	for _, indexBlob := range indexBlobs {
 		if err := bm.addIndexBlobsToBuilder(ctx, bld, indexBlob, opt); err != nil {
-			return err
+			return errors.Wrap(err, "error adding index to builder")
 		}
+
+		inputs = append(inputs, indexBlob.Metadata)
 	}
 
 	var buf bytes.Buffer
 	if err := bld.Build(&buf); err != nil {
 		return errors.Wrap(err, "unable to build an index")
 	}
 
-	compactedIndexBlob, err := bm.writePackIndexesNew(ctx, buf.Bytes())
+	compactedIndexBlob, err := bm.indexBlobManager.writeIndexBlob(ctx, buf.Bytes())
```

Comment:
A question which I had earlier but never asked - we assume that all index blobs to be compacted can be compacted into a single index blob right? Would it ever happen that we cross the limit of blob size, in case there is any?

Reply:
right now there's no limit, but we can start thinking about splitting things up when repositories get big enough.